nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,503 @@
+
+        <programlisting>
+        services.nginx.virtualHosts.''${config.services.jitsi-meet.hostName} = {
+          enableACME = true;
+          forceSSL = true;
+        };
+        </programlisting>
+      '';
+    };
+
+    prosody.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to configure Prosody to relay XMPP messages between Jitsi Meet components. Turn this
+        off if you want to configure it manually.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.jitsi-meet.jicofo.config = mapAttrs (_: v: mkDefault v) {
+      "org.jitsi.jicofo.BRIDGE_MUC" = "jvbbrewery@internal.${cfg.hostName}";
+    };
+
+    services.prosody = mkIf cfg.prosody.enable {
+      enable = mkDefault true;
+      modules = {
+        admin_adhoc = mkDefault false;
+        bosh = mkDefault true;
+        ping = mkDefault true;
+        roster = mkDefault true;
+        saslauth = mkDefault true;
+        tls = mkDefault true;
+      };
+      extraModules = [ "pubsub" ];
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
+
+        Component "conference.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- modules: muc_meeting_id muc_domain_mapper
+
+          muc_room_locking = false
+          muc_room_default_public_jids = true
+
+        Component "internal.${cfg.hostName}" "muc"
+          storage = "memory"
+          -- muc_room_cache_size = 1000
+          admins = { "focus@auth.${cfg.hostName}", "jvb@auth.${cfg.hostName}" }
+
+        Component "focus.${cfg.hostName}"
+          component_secret = os.getenv("JICOFO_COMPONENT_SECRET")
+      '';
+      virtualHosts.${cfg.hostName} = {
+        enabled = true;
+        domain = cfg.hostName;
+        extraConfig = ''
+          authentication = "anonymous"
+          c2s_require_encryption = false
+          admins = { "focus@auth.${cfg.hostName}" }
+        '';
+      };
+      virtualHosts."auth.${cfg.hostName}" = {
+        enabled = true;
+        domain = "auth.${cfg.hostName}";
+        extraConfig = ''
+          authentication = "internal_plain"
+        '';
+      };
+    };
+    systemd.services.prosody.serviceConfig = mkIf cfg.prosody.enable {
+      EnvironmentFile = [ "/var/lib/jitsi-meet/secrets" ];
+      SupplementaryGroups = [ "jitsi-meet" ];
+    };
+
+    users.groups.jitsi-meet = {};
+    systemd.tmpfiles.rules = [
+      "d '/var/lib/jitsi-meet' 0750 root jitsi-meet - -"
+    ];
+
+    systemd.services.jitsi-meet-init-secrets = {
+      wantedBy = [ "multi-user.target" ];
+      before = [ "jicofo.service" "jitsi-videobridge2.service" ] ++ (optional cfg.prosody.enable "prosody.service");
+      serviceConfig = {
+        Type = "oneshot";
+      };
+
+      script = let
+        secrets = [ "VIDEOBRIDGE_SECRET" "JICOFO_COMPONENT_SECRET" "JICOFO_USER_SECRET" ];
+      in
+      ''
+        if [ ! -f /var/lib/jitsi-meet/secrets ]; then
```

Comment:
Great work! I am really a fan of this module.

I just upgraded and ran into problems because the environment variables were renamed. I wonder if we could make it more robust by

1. checking that all required environment variables are set and if not fail with an error that indicates that you can simply delete the secrets file and execute "systemctl ..." to restart,
2. or appending missing variables to the end of the file.

The only downside I see is when the user overrides the config in a way that they don't even need the secrets. But that is probably best handled by an off switch for this check/logic since it is quite an "off-the-beaten" track config.

Ping me, if you agree and I can provide an appropriate commit based on your PR.

Reply:
I had the same problem. A bit more robustness or An indicator of what‘s wrong would be really nice.

Otherwise great module, thanks. 👍

Reply:
Added check for the required variables to the latest version. Can't promise this part won't change again since the secret distribution seems to be rather tricky.